lib/browser/growl.js

```diff
@@ -1,5 +1,212 @@
+};
+
+/**
+ * Implements browser notifications as a pseudo-reporter.
+ *
+ * @public
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/API/notification|Notification API}
+ * @see {@link https://developers.google.com/web/fundamentals/push-notifications/display-a-notification|Displaying a Notification}
+ * @see {@link Growl#isPermitted}
+ * @see {@link Mocha#_growl}
+ * @param {Runner} runner - Runner instance.
+ */
+exports.notify = function(runner) {
+  var promise = isPermitted();
+
+  /**
+   * Attempt notification.
+   */
+  var sendNotification = function() {
+    // If user hasn't responded yet... "No notification for you!" (Seinfeld)
+    Promise.race([promise, Promise.resolve(undefined)])
+      .then(canNotify)
+      .then(function() {
+        display(runner);
+      })
+      .catch(notPermitted);
+  };
+
+  runner.once('end', sendNotification);
+};
+
+/**
+ * Checks if browser notification is permitted by user.
+ *
+ * @private
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/API/Notification/permission|Notification.permission}
+ * @see {@link Mocha#growl}
+ * @see {@link Mocha#isGrowlPermitted}
+ * @returns {Promise<boolean>} promise determining if browser notification
+ *     permissible when fulfilled.
+ */
+function isPermitted() {
+  var permitted = {
+    granted: Promise.resolve(true),
+    denied: Promise.resolve(false),
+    default: (function ask() {
+      Notification.requestPermission().then(function(permission) {
+        return Promise.resolve(permission === 'granted');
+      });
+    })()
+  };
+
+  return permitted[Notification.permission];
+}
+
+/**
+ * A promise determining if notification can proceed.
+ *
+ * @promise Growl~CanNotifyPromise
+ * @fulfill {boolean} Notification allowed (<code>true</code>).
+ * @reject {Error} Notification denied (or request unacknowledged) by user.
+ */
+
+/**
+ * @summary
+ * Determines if notification should proceed.
+ *
+ * @description
+ * Notification should <strong>not</strong> proceed unless `value` is true.
+ *
+ * `value` will equal one of:
+ * <ul>
+ *   <li><code>true</code> (from `isPermitted`)</li>
+ *   <li><code>false</code> (from `isPermitted`)</li>
+ *   <li><code>undefined</code> (from `Promise.race`)</li>
+ * </ul>
+ *
+ * @private
+ * @param {Promise<boolean|undefined>} value - Determines if browser
+ *     notification permissible when fulfilled.
+ * @returns {Growl~CanNotifyPromise} Should notification proceed?
+ */
+function canNotify(value) {
+  if (!value) {
+    var why = value === false ? 'blocked' : 'unacknowledged';
+    var reason = 'not permitted by user (' + why + ')';
+    return Promise.reject(new Error(reason));
+  }
+  return Promise.resolve(true);
+}
+
+/**
+ * Displays the notification.
+ *
+ * @param {Runner} runner - Runner instance.
+ * @private
+ */
+function display(runner) {
+  var stats = runner.stats;
+  var symbol = {
+    cross: '\u274C',
+    tick: '\u2705'
+  };
+  var logo = require('../../package').logo;
```

Comment:
I'm not sure if browserify's resolution algorithm is going to be able to tell that this is the same thing as `package.json`.  can you please ensure that there aren't *two* `package.json`'s inlined in the browser bundle?

Reply:
It does. There's only one (search for "homepage" to find the _cullified_ version).